### PR TITLE
fix: Fix Ivy Failing Test: torch - manipulation.reshape

### LINKED ISSUE #28017: Fix Ivy Failing Test: torch - manipulation.reshape
ToDo List: #27501 
Sub-Task in the "Priority Open" section

### PR DESCRIPTION
**Pull request by sgalpha01**
# PR Description
Fixed Ivy Failing Test: `numpy` - `manipulation.reshape`

```
❯ pytest ivy_tests/test_ivy/test_functional/test_core/test_manipulation.py::test_reshape --no-header --no-summary --backend torch
Use Database in ReadOnly Mode with local caching !
===================================================================== test session starts =====================================================================
collected 1 item                                                                                                                                              

ivy_tests/test_ivy/test_functional/test_core/test_manipulation.py .                                                                                     [100%]

================================================================ 1 passed, 4 warnings in 5.41s ================================================================
```

## Related Issue

Closes #28017

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### Socials

Twitter: [r0gue_shinobi](https://twitter.com/r0gue_shinobi)

### CODE REVIEW

**Comment by sgalpha01** [comment]
I have a few questions regarding to the `ivy/functional/backends/torch/manipulation.py` file. The use of `copy` parameter is inconsistent across the file. For example, in some functions, it is 
```python
    if copy:
        x = x.clone().detach()
```
whereas in some other, it is
```python
        if copy:
            newarr = torch.clone(x)
            return [newarr]
```

Notice the style of variable naming. It is not an issue functionality-wise. Other doubt, what should be the behaviour when `copy=True`? In some functions, the `tensor` is copied with `detach()` method which removes it from the computation graph, but in other, it is not called.

**Comment by joaozenobio** [comment]
Hi @sgalpha01 ! THanks for the PR submitted :smile: pin me whenever you are ready for a review.

> I have a few questions regarding to the `ivy/functional/backends/torch/manipulation.py` file. The use of `copy` parameter is inconsistent across the file. For example, in some functions, it is
> 
> ```python
>     if copy:
>         x = x.clone().detach()
> ```
> 
> whereas in some other, it is
> 
> ```python
>         if copy:
>             newarr = torch.clone(x)
>             return [newarr]
> ```
> 
> Notice the style of variable naming. It is not an issue functionality-wise. Other doubt, what should be the behaviour when `copy=True`? In some functions, the `tensor` is copied with `detach()` method which removes it from the computation graph, but in other, it is not called.

Regarding this error, you are right, detach should not be done after the copy, this might lead to inconsistencies for which device running the function. Thank you for pointing out! :rocket:

**Comment by sgalpha01** [comment]
Hey @joaozenobio, thanks for the review. I think now it is good to go. I removed the `.detach()` method from the previous code also.

**Comment by joaozenobio** [comment]
Hi @sgalpha01 ! Awsome, tests are passing and the fixes are great. Nice work, I will now merge your PR :rocket:
